packages/extensions/openapi-to-typespec/src/generate/generate-service-information.ts

```diff
@@ -91,3 +92,17 @@ function getEndpointParameters(endpoint: string) {
 
   return params;
 }
+
+function generateUseAuth(auth: Auth | undefined, statements: string[]): void {
+  if(!auth) {
+    return;
+  }
+  if(auth.kind === "AadOauth2Auth") {
+    const scopes = `[${auth.scopes.map(s => `"${s}"`).join()}]`
+    statements.push(`@useAuth(AadOauth2Auth<${scopes}>)`);
```

Comment:
What happens if there's an auth url specified?

Reply:
Also will this support definitions like the following: https://github.com/Azure/azure-rest-api-specs/blob/e69839a006ffc8902412cc38228fd600942c2f73/specification/cognitiveservices/ContentSafety/main.tsp#L17-L22 ?

Reply:
It will but it wouls generate 2 @useAuth decorators instead of the union

Reply:
Multiple auth decorators are not supported :/ Only the first one would be used. Here is the issue I opened for the linter to warn against having multiple useAuth decorators: https://github.com/microsoft/typespec/issues/2787

Reply:
We'd need to do the | syntax when generating

Reply:
Great! Thanks for catching this :)